lib/views/topLevelView.js

```diff
@@ -0,0 +1,351 @@
+        section: typeShortName,
+        user: req.user
+    }
+
+    var meta
+    var sbol
+    var topLevel
+    var collectionIcon 
+    var remote
+
+    var collections = []
+    var submissionCitations = []
+    var citations = []
+    var builds = []
+
+    var encodedProteins = []
+    var otherComponents = []
+    var mappings = {}
+
+    const { graphUri, uri, designId, share, url } = getUrisFromReq(req, res)
+
+    var templateParams = {
+        uri: uri
+    }
+
+    var getCitationsQuery = loadTemplate('sparql/GetCitations.sparql', templateParams)
+
+    var getBuildsQuery = loadTemplate('sparql/GetImplementations.sparql', templateParams)
+
+    fetchSBOLObjectRecursive(uri, graphUri).then((result) => {
+	
+        sbol = result.sbol
+        topLevel = result.object
+        remote = result.remote
+
+        if(!topLevel || topLevel instanceof URI) {
+            locals = {
+                config: config.get(),
+                section: 'errors',
+                user: req.user,
+                errors: [ uri + ' Record Not Found: ' + topLevel ]
+            }
+            res.send(pug.renderFile('templates/views/errors/errors.jade', locals))
+            return Promise.reject()
+        }
+	meta = summarizeSBOL(typeShortName,topLevel,req,sbol,remote,graphUri)
+	
+    }).then(function lookupCollections() {
+
+        return Promise.all([
+	    getContainingCollections(uri, graphUri, req.url).then((_collections) => {
+
+		collections = _collections
+
+		collections.forEach((collection) => {
+
+		    collection.url = uriToUrl(collection.uri)
+
+                    const collectionIcons = config.get('collectionIcons')
+                    
+                    if(collectionIcons[collection.uri])
+			collectionIcon = collectionIcons[collection.uri]
+		})
+            }),
+
+	    sparql.queryJson(getCitationsQuery, graphUri).then((results) => {
+
+                citations = results
+
+            }).then(() => {
+
+                return retrieveCitations(citations).then((resolvedCitations) => {
+
+                    submissionCitations = resolvedCitations;
+
+                })
+
+            }),
+
+            sparql.queryJson(getBuildsQuery, graphUri).then((results) => {
+
+		builds = results
+		builds.forEach((build) => {
+		    build.url = uriToUrl(build.impl)
+		})
+            })
+
+        ])
+
+    }).then(function lookupEncodedProteins() {
+
+        var query =
+            'PREFIX sybio: <http://w3id.org/sybio/ont#>\n' +
+        'SELECT ?subject WHERE {' +
+            '   ?subject sybio:encodedBy <' + uri + '>' +
+            '}'
+
+        return sparql.queryJson(query, graphUri).then((results) => {
+
+            encodedProteins = results.map((result) => {
+                return result.subject
+            })
+
+        })
+
+    }).then(function getOtherComponentMetaData() {
+
+        if(meta.protein && meta.protein.encodedBy)
+            otherComponents = otherComponents.concat(meta.protein.encodedBy)
```

Comment:
if you still have todo here then wrap the if contents in {}

Reply:
Would like to move these things out of this file and wrap with only if CD.  Not sure how.